model/modelprocessor/datastream.go

```diff
@@ -64,21 +67,30 @@ func (s *SetDataStream) setDataStream(event *modelpb.APMEvent) {
 	switch event.Type() {
 	case modelpb.SpanEventType, modelpb.TransactionEventType:
 		event.DataStream.Type = tracesType
-		event.DataStream.Dataset = tracesDataset
-		// In order to maintain different ILM policies, RUM traces are sent to
-		// a different datastream.
-		if isRUMAgentName(event.GetAgent().GetName()) {
-			event.DataStream.Dataset = rumTracesDataset
+		if event.DataStream.Dataset == "" {
+			// Only set dataset if it is not already set in the input event
+			event.DataStream.Dataset = tracesDataset
+			// In order to maintain different ILM policies, RUM traces are sent to
+			// a different datastream.
+			if isRUMAgentName(event.GetAgent().GetName()) {
```

Comment:
[For reviewer] RUM check is bypassed if DataStream is specified. Is this desirable?

Reply:
This reminded me: RUM agents shouldn't be control routing until we have some way for operators to restrict them. Otherwise (unauthenticated, untrusted) RUM agents can create arbitrarily many data streams, and DoS Elasticsearch.